### PR TITLE
Doccano 1.7 compatibility update

### PR DESCRIPTION
**Pull request by lance132**
Adds missing tests for span, span-type, relation, relation-type
Adds category and category-type controllers for Document Classification projects plus corresponding tests
Adds text controller for Seq2seq projects plus corresponding tests
Updates api endpoints for labels and comments
Fixes bug with project attributes

### CODE REVIEW

File: doccano_client/beta/controllers/comment.py [hunk 1]
```diff
@@ -48,7 +48,12 @@ def __init__(self, parent_url: str, client_session: Session) -> None:
     @property
     def comments_url(self) -> str:
         """Return an api url for comments list of an object"""
-        return f"{self._parent_url}/comments"
+        if "/examples" in self._parent_url:
```
Comment (houssam7737) [inline doccano_client/beta/controllers/comment.py:51]: A quick comment here might be helpful for future contributors in explaining why we have this branching here which we didn't have before

File: doccano_client/beta/controllers/comment.py [hunk 2]
```diff
@@ -57,20 +62,27 @@ def all(self) -> Iterable[CommentController]:
...
-                comment=Comment(**sanitized_comment_dict),
-                username=comment_dict["username"],
-                created_at=comment_dict["created_at"],
-                example=comment_dict["example"],
-                id=comment_dict["id"],
-                comments_url=self.comments_url,
-                client_session=self.client_session,
-            )
+
+        while True:
```
Comment (houssam7737) [inline doccano_client/beta/controllers/comment.py:66]: Quick comment here would be helpful explaining why we have a while true loop, explaining that we don't get all comments back at once and need to call the next url sent back by django

File: doccano_client/beta/controllers/category.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                id=category_dict["id"],
+                categories_url=self.categories_url,
+                client_session=self.client_session,
+            )
+
+    def create(self, category: Category) -> CategoryController:
+        """Create a new category, return the generated controller
+
+        Args:
+            category: Category. See mock_api_responses for the desired format.
```
Comment (houssam7737) [inline doccano_client/beta/controllers/category.py:73]: If it's not too much to ask might good to add the format here quickly, saving future contributors/readers of code from having to look at another file.

File: doccano_client/beta/controllers/example.py [hunk 4]
```diff
@@ -126,8 +138,7 @@ def create(self, example: Example) -> ExampleController:
         """Upload new example for Doccano project, return the generated controller
 
         Args:
-            example: Example. The only fields that will be uploaded are text, annnotations,
-                and meta.
+            example: Example. See mock_api_responses for the desired format.
```
Comment (houssam7737) [inline doccano_client/beta/controllers/example.py:141]: Same as previously: If it's not too much to ask might good to add the format here quickly, saving future contributors/readers of code from having to look at another file.

File: doccano_client/beta/controllers/relation.py [hunk 5]
```diff
@@ -66,3 +66,26 @@ def all(self) -> Iterable[RelationController]:
                 relations_url=self.relations_url,
                 client_session=self.client_session,
             )
+
+    def create(self, relation: Relation) -> RelationController:
+        """Create a new relation, return the generated controller
+
+        Args:
+            relation: Relation. See mock_api_responses for the desired format.
```
Comment (houssam7737) [inline doccano_client/beta/controllers/relation.py:74]: If it's not too much to ask might good to add the format here quickly, saving future contributors/readers of code from having to look at another file.

File: doccano_client/beta/controllers/span.py [hunk 6]
```diff
@@ -68,7 +68,7 @@ def create(self, span: Span) -> SpanController:
         """Create a new span, return the generated controller
 
         Args:
-            span: Span. The only fields that will be uploaded are text, annnotations, and meta.
+            span: Span. See mock_api_responses for the desired format.
```
Comment (houssam7737) [inline doccano_client/beta/controllers/span.py:71]: Here as well. If it's not too much to ask might good to add the format here quickly, saving future contributors/readers of code from having to look at another file.

File: doccano_client/beta/controllers/text.py [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+                id=text_dict["id"],
+                texts_url=self.texts_url,
+                client_session=self.client_session,
+            )
+
+    def create(self, text: Text) -> TextController:
+        """Create a new text, return the generated controller
+
+        Args:
+            text: Text. See mock_api_responses for the desired format.
```
Comment (houssam7737) [inline doccano_client/beta/controllers/text.py:71]: And here.

**Comment by houssam7737** [comment]
Overall looks very solid with a great test suite. A couple minor suggestions to improve documentation and I think this should be ready to merge! Thanks for your contribution! (with your update the client should now be fully operational with Doccano 1.7)

**Comment by lance132** [comment]
Addressed the comments!

**Review by houssam7737 — APPROVED** [review]
Looks good to me now! Thanks

**Pull request merged**